update.sh

```diff
@@ -14,5 +19,23 @@ fi
 git submodule sync
 git submodule update --init --recursive
 
-git submodule foreach python setup.py develop --user
-git submodule foreach pip install --user --editable .
+for submodule in core web-api france parsers
+do
+	cd openfisca/$submodule/
+	python setup.py develop $user_option
+	pip install $user_option  --editable .
+	cd ../..
+done
+
+for old_symlink_extension in $(find openfisca/france/openfisca_france/model/extensions -type l)
+do
+	rm $old_symlink_extension
+done
+
+for extension in $(ls  -d openfisca/extensions/*)
+do
+	if [[  $(git submodule | grep $extension) ]]
+	then
+		ln -s $(pwd)/$extension openfisca/france/openfisca_france/model/extensions/
+	fi
+done
```

Comment:
Wouldn't it be simpler to have a single iteration loop over the submodules, check whether they are in the `openfisca/extensions` directory and branch to either `pip install` or `ln -s`? You can even put each treatment in its own function to make it clearer.